src/App.vue

```diff
@@ -424,6 +487,197 @@ export default {
     ]),
     ...mapActions("selectors", ["hideSelector"]),
     ...Utilities,
+    /**
+     * NOTE - In this method, "org" will be used in place of "workspace"
+     * to keep variable names short and readable.
+     *
+     * This method merges the user's and org's settings.
+     * This is needed to show both org level and user level settings in one settings menu.
+     * The global default settings object is used as a structure of the keys to iterate on.
+     * While iterating on the keys of the global default settings object, these rules are followed to merge
+     * - If a key is not present in org's settings, use the key from user's settings and skip to next key
+     * - If a key is present in org's settings, use the scope for that key and move to it's children
+     * - The above process is done for headers, tabs and atomic settings.
+     * - For the lowest level keys (atomic settings), use org's setting value if available otherwise use user's setting value
+     *
+     * @param {Object} userSettings - User's version of settings
+     * @param {Object} orgSettings - Org's version of settings
+     * @returns {Object} An object with user's and org's settings merged (with org's settings taking priority)
+     */
+    mergeSettings(userSettings, orgSettings) {
```

Comment:
the fact that it is merging is also an implementation detail that the function should abstract away. (name of the function could be something else too)
```suggestion
    prepareUserSettingsForRendering(userSettings, orgSettings) {
```

Reply:
Where do I abstract away this?
Things are abstracted away from user interface or from a library we're using. But how and where do I abstract away this implementation? 

And I think mergeSettings name explains what's happening right?
Also I think `prepareUserSettingsForRendering` is wrong because we're not preparing "user" settings.
We're using "user" + "workspace" settings and merging them into one to create "settingsToRender"